tools/infrasys/src/error.rs

```diff
@@ -0,0 +1,165 @@
+use snafu::Snafu;
+use std::io;
+use std::path::PathBuf;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub(super)")]
+pub enum Error {
+    #[snafu(display(
+        "Failed to create CFN stack '{}' in '{}': {}",
+        stack_name,
+        region,
+        source
+    ))]
+    CreateStack {
+        stack_name: String,
+        region: String,
+        source: rusoto_core::RusotoError<rusoto_cloudformation::CreateStackError>,
+    },
+
+    #[snafu(display(
+        "Recieved CREATE_FAILED status for CFN stack '{}' in '{}'",
+        stack_name,
+        region
+    ))]
+    CreateStackFailure { stack_name: String, region: String },
+
+    #[snafu(display("Error splitting shell command '{}': {}", command, source))]
+    CommandSplit {
+        command: String,
+        source: shell_words::ParseError,
+    },
+
+    #[snafu(display("Error reading Infra.toml: {}", source))]
+    Config { source: pubsys_config::Error },
+
+    #[snafu(display(
+        "Stuck in indefinite CREATE_IN_PROGRESS loop for CFN stack '{}' in '{}'",
+        stack_name,
+        region
+    ))]
+    CreateStackTimeout { stack_name: String, region: String },
+
+    #[snafu(display(
+        "Failed to fetch stack details for CFN stack '{}' in '{}': {}",
+        stack_name,
+        region,
+        source
+    ))]
+    DescribeStack {
+        stack_name: String,
+        region: String,
+        source: rusoto_core::RusotoError<rusoto_cloudformation::DescribeStacksError>,
+    },
+
+    #[snafu(display("Missing environment variable '{}'", var))]
+    Environment {
+        var: String,
+        source: std::env::VarError,
+    },
+
+    #[snafu(display("File already exists at '{}'", path.display()))]
+    FileExists { path: PathBuf },
+
+    #[snafu(display("Failed to open file at '{}': {}", path.display(), source))]
+    FileOpen { path: PathBuf, source: io::Error },
+
+    #[snafu(display("Failed to read file at '{}': {}", path.display(), source))]
+    FileRead { path: PathBuf, source: io::Error },
+
+    #[snafu(display("Failed to write file in '{}': {}", path.display(), source))]
```

Comment:
Is this path a directory or the actual file path that failed to write"?

```suggestion
    #[snafu(display("Failed to write file at '{}': {}", path.display(), source))]
```